packages/devtools_app/lib/src/performance/event_details.dart

```diff
@@ -134,81 +134,172 @@ class EventSummary extends StatelessWidget {
+                fit: FlexFit.tight,
+                child: _buildDataItem(
+                  context: context,
+                  title: 'Thread id',
+                  inlineValue: '${firstTraceEvent.threadId}',
+                ),
+              ),
+              Flexible(
+                fit: FlexFit.tight,
+                child: _buildDataItem(
+                  context: context,
+                  title: 'Process id',
+                  inlineValue: '${firstTraceEvent.processId}',
+                ),
+              ),
+            ],
+          ),
         ),
-        if (event.isAsyncEvent) _asyncIdTile(),
-        if (_connectedEvents.isNotEmpty) _buildConnectedEvents(),
-        if (_eventArgs.isNotEmpty) _buildArguments(),
+        if (_connectedEvents.isNotEmpty) _buildConnectedEvents(context),
+        if (_eventArgs.isNotEmpty) _buildArguments(context),
       ],
     );
   }
 
-  Widget _asyncIdTile() {
+  Widget _asyncIdTile(BuildContext context) {
     String asyncId;
     if (event is OfflineTimelineEvent) {
       asyncId = event.traceEvents.first.event.id;
     } else {
       asyncId = (event as AsyncTimelineEvent).asyncId;
     }
-    return ListTile(
-      title: const Text('Async id'),
-      subtitle: Text(asyncId),
+    return _buildDataItem(
+      context: context,
+      title: 'Async id',
+      inlineValue: asyncId,
     );
   }
 
-  Widget _buildConnectedEvents() {
-    return ExpansionTile(
-      title: const Text('Connected events'),
+  Widget _buildConnectedEvents(BuildContext context) {
+    return _buildExpandingDataItem(
+      context: context,
+      title: 'Connected events',
       children: [
-        Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: [
-            for (var e in _connectedEvents) _buildConnectedEvent(e),
-          ],
-        ),
+        for (var e in _connectedEvents) _buildConnectedEvent(context, e),
       ],
     );
   }
 
-  Widget _buildConnectedEvent(TimelineEvent e) {
+  Widget _buildConnectedEvent(BuildContext context, TimelineEvent e) {
     final eventArgs = {
       'startTime': msText(e.time.start - event.time.start),
       'args': e.traceEvents.first.event.args,
     };
-    return ListTile(
-      title: Text(e.name),
-      subtitle: FormattedJson(json: eventArgs),
+    return _buildDataItem(
+      context: context,
+      title: e.name,
+      child: FormattedJson(
+        json: eventArgs,
+        useSubtleStyle: true,
+      ),
     );
   }
 
-  Widget _buildArguments() {
-    return ExpansionTile(
-      title: const Text('Arguments'),
+  Widget _buildArguments(BuildContext context) {
+    return _buildExpandingDataItem(
+      context: context,
+      title: 'Arguments',
       children: [
-        ListTile(
-          subtitle: FormattedJson(json: _eventArgs),
+        FormattedJson(
+          json: _eventArgs,
+          useSubtleStyle: true,
         ),
       ],
     );
   }
+
+  Widget _buildDataItem({
```

Comment:
nit: add a TODO to make this and other widget like methods in this class widgets.
I think a good rule is that if a function takes a BuildContext, it should probably be a Widget.

Reply:
went ahead and converted to stateless widgets in this PR